j-lawyer-client/src/com/jdimension/jlawyer/client/editors/files/InvolvedPartyEntryPanel.java

```diff
@@ -1455,6 +1483,54 @@ private void mnuCopyActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST
         }
     }//GEN-LAST:event_mnuCopyActionPerformed
 
+    private void mnuFindRouteToAddressActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_mnuFindRouteToAddressActionPerformed
+        if (this.a != null) {
+            
+            // PROFILE as startAdress
+            ServerSettings sset = ServerSettings.getInstance();
+            String companyCity = sset.getSetting(ServerSettings.PROFILE_COMPANYCITY, "");
+            String companyStreet = sset.getSetting(ServerSettings.PROFILE_COMPANYSTREET, "");
+            String companyZip = sset.getSetting(ServerSettings.PROFILE_COMPANYZIP, "");
+            
+            String startAddress = companyCity + "+" + companyZip + "+" + companyStreet.replace(" ", "+");
+            String destinationAddress = a.getCity() + "+" + a.getZipCode() + "+" + a.getStreet() + "+" + a.getStreetNumber();
+            
+            // JOptionPane.showMessageDialog(null, startAddress);  
+            
+            try {
+                
+                String coordinates1 = FindRouteToAddress.getCoordinates(startAddress);
+                String coordinates2 = FindRouteToAddress.getCoordinates(destinationAddress);
+   
+                if (coordinates1 != null && coordinates2 != null) {
+                    
+                    FindRouteToAddress.openBrowserWithRoute(coordinates1, coordinates2);
+                    
+                } else {
+                    JOptionPane.showMessageDialog(null, "Die Koordinaten für eine oder beide Adressen konnten nicht abgerufen werden.");
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }           
+        }
+    }//GEN-LAST:event_mnuFindRouteToAddressActionPerformed
+
+    private void mnuFindAddressActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_mnuFindAddressActionPerformed
+        if (this.a != null) {
+            
+            String address = a.getCity() + "+" + a.getZipCode() + "+" + a.getStreet() + "+" + a.getStreetNumber();
+            
+            try {
+                
+                    FindRouteToAddress.openBrowserWithAddress(address);
+                
+            } catch (Exception e) {
+                JOptionPane.showMessageDialog(null, "Die Addresse konnte nicht abgerufen werden.");
+                e.printStackTrace();
```

Comment:
we should never use printstacktrace in production code, because the output will just disappear / never show up in the clients log file, which makes support and troubleshooting hard.

use this instead:

`log.error("Unable to determine address in OSM: " + address, e);`